flink-connector-jdbc-core/src/main/java/org/apache/flink/connector/jdbc/core/datastream/source/reader/extractor/ResultExtractor.java

```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package org.apache.flink.connector.jdbc.core.datastream.source.reader.extractor;
```

Comment:
This is a public interface we should not move it so easy.
If we are moving it we need to keep an class (deprecated) in origin that extends the moved one to keep retro compatibility, but I would avoid database as is the API for connectors (the things that connectors need to implement)

I understand that this refactor is because of the new implementation, but being table using the datastream why don't we put the new clase on the current package? that is a valid implementation also for datastream and useful in some cases (reading from jdbc and sending to paimon/iceberg as example)